apis/hive/v1/hiveconfig_types.go

```diff
@@ -53,6 +53,11 @@ type HiveConfigSpec struct {
 	// +optional
 	FailedProvisionConfig FailedProvisionConfig `json:"failedProvisionConfig,omitempty"`
 
+	// AWSServiceProviderConfig is used to configure setting related to being a service provider on
+	// AWS. This includes setting like credentials to become a service provider for AWS clusters.
+	// +optional
+	AWSServiceProviderConfig AWSServiceProviderConfig `json:"awsServiceProviderConfig,omitempty"`
```

Comment:
`ServiceProviderConfig` is going to be a little too general "any configuration that comes from being a service provider on the cloud platform" not sure if that will not become dump of everything. If we were to go this route what criteria should be decide to accept or reject fields in this section?

Reply:
I think we've got that problem regardless even within AWSServiceProviderConfig. Would ServiceProviderCredentialsConfig help? ServiceProviderAuthConfig? 

AWSNextThing top level struct is what I'd most like to avoid.

Reply:
updated to ServiceProviderCredentialsConfig